src/algorithms/calorimetry/HEXPLIT.cc

```diff
@@ -0,0 +1,145 @@
+// Copyright (C) 2023 Sebouh Paul
+// SPDX-License-Identifier: LGPL-3.0-or-later
+
+// References:
+//   https://arxiv.org/abs/2308.06939
+
+
+#include <DD4hep/Readout.h>
+#include <Evaluator/DD4hepUnits.h>
+#include <TInterpreter.h>
+#include <TInterpreterValue.h>
+#include <edm4hep/Vector3d.h>
+#include <Eigen/Dense>
+#include <fmt/format.h>
+#include <map>
+#include <set>
+#include <sstream>
+#include <stdexcept>
+#include <string>
+#include <tuple>
+#include <utility>
+#include <vector>
+#include <iostream>
+
+#include "HEXPLIT.h"
+
+#include "algorithms/calorimetry/HEXPLITConfig.h"
+
+//using namespace edm4eic;
+
+namespace eicrecon {
+
+void HEXPLIT::init(const dd4hep::Detector* detector, std::shared_ptr<spdlog::logger>& logger) {
+    m_log = logger;
+    m_detector = detector;
+}
+
+std::unique_ptr<edm4eic::CalorimeterHitCollection> HEXPLIT::process(const edm4eic::CalorimeterHitCollection &hits){
+  int nhits=hits.size();
+  double sl=m_cfg.side_length/dd4hep::mm;
+  double layer_spacing=m_cfg.layer_spacing/dd4hep::mm;
+  double MIP=m_cfg.MIP/dd4hep::GeV;
+  double Emin=m_cfg.Emin_in_MIPs*MIP/dd4hep::GeV;
+  double tmax=m_cfg.tmax/dd4hep::ns;
+
+
+  double x[nhits];
+  double y[nhits];
+  double z[nhits];
+  double E[nhits];
+  double t[nhits];
```

Comment:
This is probably better solved with a std::vector of tuples, which doesn't require using `nhits` to create an array.

Reply:
fixed in commit 125559e.